client/src/components/ArticleList.jsx

```diff
@@ -4,11 +4,25 @@ import Article from './Article.jsx';
 class ArticleList extends Component {
   constructor(props) {
     super(props);
+    this.state = {
+      articles: this.props.articles
+    }
+    this.sortArticles = this.sortArticles.bind(this);
+  }
+
+  componentDidMount() {
+    this.sortArticles();
+  }
+
+  sortArticles() {
+    let articles = this.state.articles;
+    let sorted = articles.sort((a, b) => a.date < b.date);
+    this.setState({articles: sorted});
```

Comment:
`.sort()` sorts the array in place, and `articles` is a reference to the articles in state, so it will actually sort the articles in `this.state.articles`. a good way to copy a state array without a direct reference is
`let articles = [...this.state.articles];`
then you can just run `articles.sort()` without reassigning it